core/src/typecheck/mod.rs

```diff
@@ -2857,7 +2900,8 @@ fn instantiate_foralls(
         ..
     } = ty
     {
-        let kind = (&var_kind).into();
+        let kind: VarKindDiscriminant = (&var_kind).into();
```

Comment:
Cosmetic change: I find it better to put an explicit type annotation when there's an `into()` that isn't used right away.